### DISCUSSION TITLE
Cloud Firestore Backend: Backend didn't respond within 10 seconds

### DISCUSSION THREAD

**Discussion by verai17** [angular/angularfire#3183] [Q&A]
I am currently working on a client application and using angularfire for firebase integration. The application works fine on my local machine except that when deployed to the AWS server, the application would received this the error `Cloud Firestore Backend: Backend didn't respond within 10 seconds`. I have done many research on how to fixed this error, yet the error still persist. Please see below details. 

Client Project:
Angular `v12.2.11`
Npm `v6.14.12`
Node `v12.22.1`

AWS Server:
Memory : `1024`
CPU : `256`

Solutions (tried and didn't work):
- Arrange configurations at app.module based on documentation
Reference: https://github.com/angular/angularfire/blob/master/docs/install-and-setup.md
 
- Added Firebase Auth Persistence ‘NONE’

- Downgrade angularFire from into v7.2 to v7.0 with firebase v9.6
- Downgrade angularFire from into v7.0 to v6.1, and, Firebase from v9.6 to v8.10
- Downgrade Firebase from v8.10 to v7.24
Reference: https://github.com/angular/angularfire

 
Hope someone from the community can help me out since its been weeks trying to resolve the issue. 🙏🏻

**Comment by rusudinu** [comment]
You can try to verify if the AWS server has outbound access to the Firebase backend and if there are any firewall or security groups that might be blocking the connection. Also, how close are the regions of the two services?

**Comment by jainut20** [comment]
Try setting experimentalForceLongPolling to True
